tests/lib/rules/require-tagless-components.js

```diff
@@ -0,0 +1,100 @@
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/require-tagless-components');
+const RuleTester = require('eslint').RuleTester;
+
+const { ERROR_MESSAGE_DEFAULT_TAGNAME, ERROR_MESSAGE_TAGNAME_NOT_EMPTY } = rule;
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('babel-eslint'),
+  parserOptions: { ecmaVersion: 6, sourceType: 'module' },
+});
+
+ruleTester.run('require-tagless-components', rule, {
+  valid: [
+    `
+      import Component from '@ember/component';
+      export default Component.extend({ tagName: '' });
+    `,
+    `
+      import Component from '@ember/component';
+      export default class MyComponent extends Component {
+        tagName = ''
+      }
+    `,
+    `
+      import Component from '@glimmer/component';
+      export default class MyComponent extends Component {}
+    `,
+  ],
+  invalid: [
+    {
+      code: `
+        import Component from '@ember/component';
+        export default Component.extend({});
+      `,
+      output: null,
+      errors: [{ message: ERROR_MESSAGE_DEFAULT_TAGNAME, line: 3 }],
```

Comment:
Can you add `type` to each of these error objects to ensure the violation is on the right element?